### PR TITLE
Add switch field dropdown on EditableField header

### PR DESCRIPTION
**Pull request by rlecellier**
pull request for proposed issue https://github.com/Kinto/formbuilder/issues/153

### CODE REVIEW

**Comment by rlecellier** [comment]
I'm not sure about EditableField.js:
```
function pickKeys(source, target) {
  const result = {};
  for (let key in source) {
    if (key =="type") {
      // type dont have to be update from form data.
      // it break switchField action.
      continue;
    }
    result[key] = target[key];
  }
  return result;
}
```

i also fix this by adding a callback 
```
handleSwitchField(newFieldType) {
  const forcedProperties = {
    type: newFieldtype
  };
  this.setState(editedSchema:{...this.state.editedSchema, ...forcedProperties);
} 
```

I finaly felt that skyping type was the easiest.

File: formbuilder/components/builder/EditableField.js [hunk 1]
```diff
@@ -2,11 +2,18 @@ import React, { Component } from "react";
...
+import { ButtonToolbar, Button} from "react-bootstrap";
+import FieldListDropdown from "./FieldListDropdown"
 
 
 function pickKeys(source, target) {
   const result = {};
   for (let key in source) {
+    if (key =="type") {
+      // type dont have to be update from form data.
+      // it break switchField action.
```
Comment (almet) [inline formbuilder/components/builder/EditableField.js]: Unfortunately I don't understand the comment you entered here. Can you rephrase? Thanks!

File: formbuilder/components/builder/FieldListDropdown.js [hunk 2]
```diff
@@ -5,34 +5,43 @@ import {Dropdown, MenuItem}  from "react-bootstrap"
...
+    };
   }
 
-  handleAddField(fieldIndex, event) {
+  handleFieldListAction(fieldIndex, event) {
     const fieldList = this.state.fieldList;
+
     fieldIndex = parseInt(fieldIndex, 10);
 
     if (typeof fieldList[fieldIndex] != "undefined") {
```
Comment (almet) [inline formbuilder/components/builder/FieldListDropdown.js]: `!==`

Comment (almet) [inline formbuilder/components/builder/FieldListDropdown.js]: `!==`

File: formbuilder/components/builder/FieldListDropdown.js [hunk 3]
```diff
@@ -5,34 +5,43 @@ import {Dropdown, MenuItem}  from "react-bootstrap"
...
-      this.props.addField(fieldList[fieldIndex]);
+      const field = fieldList[fieldIndex];
+
+      if (this.props.currentFieldName !== false) {
```
Comment (almet) [inline formbuilder/components/builder/FieldListDropdown.js]: It's not clear to me here what it means when `currentfieldName` is set to `false`. We probably should use an intermediary variable here to make it clearer.

File: formbuilder/components/builder/FieldListDropdown.js [hunk 4]
```diff
@@ -5,34 +5,43 @@ import {Dropdown, MenuItem}  from "react-bootstrap"
...
     }
   }
 
   render () {
     return (
-      <Dropdown title="Add a field" dropup id="split-button-dropup" className={this.props.className}>
-        <Dropdown.Toggle>
-          <i className="glyphicon glyphicon-plus" />
-          Add a field
+      <Dropdown dropup={this.props.currentFieldName == false} id="split-button-dropup" className={this.props.className}>
```
Comment (almet) [inline formbuilder/components/builder/FieldListDropdown.js]: Once we have the intermediary variable, this would be replaced with it.

**Comment by almet** [comment]
Hey @rlecellier , any news?

**Comment by rlecellier** [comment]
I'm reading doc / working on my server upgrade, cf [your infra](https://github.com/almet/infra) and my recent [resume](https://notehub.org/yf670) of it.

I'll finish this just after.

File: formbuilder/components/builder/EditableField.js [hunk 5]
```diff
@@ -39,16 +48,23 @@ class FieldPropertiesEditor extends Component {
...
-            close <i className="glyphicon glyphicon-remove-sign"/>
-          </button>
-          <button type="button" className="close-btn" onClick={onDelete} aria-label="Delete">
-            delete <i className="glyphicon glyphicon-trash"/>
-          </button>
+        <div className="panel-heading clearfix">
+            <strong className="panel-title">Edit {name}</strong>
+
+            <ButtonToolbar className="pull-right">
+              <FieldListDropdown bsStyle="link" {...this.props} currentFieldName={this.props.name}>
```
Comment (almet) [inline formbuilder/components/builder/EditableField.js]: Instead of doing these tests to see if the state changed, I believe a simpler way to do would be to detect when the current field changes and trigger a different action, that would save us from a lot of if checks and special cases here and there.

Comment (rlecellier) [inline formbuilder/components/builder/EditableField.js]: I squash everything and update the review.
I don't see how we can observe the type change. The type is string(text, textarea, select, radio) or boolean(checkbox).
The comment on my previous request was wrong (fixed).

**Comment by almet** [comment]
See #167
